wgpu-core/src/device/life.rs

```diff
@@ -248,21 +332,29 @@ impl<A: HalApi> LifetimeTracker<A> {
             return;
         }
 
-        for buffer in self.mapped.drain(..) {
-            let submit_index = buffer.submission_index();
+        // To avoid a borrow checker conflict between the mutable borrow of `self.mapped`
+        // and get_buffer_latest_submission_index, we steal the vector, drain it, then put it back.
+        //
+        // This way we still re-use the memory allocation, without the conflict.
+        let mut mapped = mem::take(&mut self.mapped);
+        for buffer in mapped.drain(..) {
+            let submit_index = self.get_buffer_latest_submission_index(&buffer);
             log::trace!(
                 "Mapping of {} at submission {:?} gets assigned to active {:?}",
                 buffer.error_ident(),
                 submit_index,
-                self.active.iter().position(|a| a.index == submit_index)
+                self.active
+                    .iter()
+                    .position(|a| Some(a.index) == submit_index)
             );
 
             self.active
                 .iter_mut()
-                .find(|a| a.index == submit_index)
+                .find(|a| Some(a.index) == submit_index)
```

Comment:
```suggestion
                .rev()
                .find(|a| a.contains_buffer(&buffer))
```

Since we already iterate through the submissions I think we can do this instead. We don't need to `mem::take` then either.